clang/lib/CIR/Dialect/IR/CIRDialect.cpp

```diff
@@ -2537,6 +2537,17 @@ LogicalResult cir::GetMemberOp::verify() {
   return mlir::success();
 }
 
+//===----------------------------------------------------------------------===//
+// GetElementOp Definitions
+//===----------------------------------------------------------------------===//
+
+LogicalResult cir::GetElementOp::verify() {
+  auto arrayTy = mlir::cast<cir::ArrayType>(getBaseType().getPointee());
+  if (getElementType() != arrayTy.getElementType())
+    return emitError() << "element type mismatch";
+  return mlir::success();
+}
```

Comment:
I believe this can be written directly in ODS as:

```
  TypesMatchWith<"type of 'result' matches element type of 'array'",
    "array", "result", "mlir::cast<cir::ArrayType>($_self).getElementType()">
```

we might want to create a helper trait for this as it happens a lot in CIR, e.g., `VecExtractOp`.

Reply:
This will also infer the result type, so it is not needed to be specified in the assembly form.

Reply:
It was a bit harder than this, but I managed to use a `TypesMatchWith`.